### PR TITLE
[Issue 157] Support report layout

### LINKED ISSUE #157: Support report layout
https://github.com/jupyter-incubator/dashboards/issues/203 will introduce a simpler dashboard "report" layout:

> **Report mode:** a simple view which shows cells in notebook order, single column and full width. The only action available to the user is whether to show/hide each cell.

We should support this layout once it is implemented in the dashboards notebook extension.

### PR DESCRIPTION
**Pull request by dalogsdon**
Resolves #157

### CODE REVIEW

**Comment by parente** [comment]
Hold until jupyter-incubator/dashboards#211 goes in.

File: etc/notebooks/test/test_style_match.ipynb [hunk 1]
```diff
@@ -257,7 +257,7 @@
     }
    },
    "source": [
-    "<img src=\"http://www.bigcat.com/CZ2/cougar2.jpg\">"
+    "<img src=\"http://cincinnatizoo.org/wp-content/uploads/2011/02/cougar1.jpg\">"
```
Comment (parente) [inline etc/notebooks/test/test_style_match.ipynb]: This was bothering me. Thank you.

File: routes/renderers.js [hunk 2]
```diff
@@ -85,12 +85,12 @@ function _renderDashboard(req, res, next, opts) {
...
-                return ( currentCell.hasOwnProperty('metadata') &&
-                         currentCell.metadata.hasOwnProperty('urth') &&
-                         currentCell.metadata.urth.hasOwnProperty('dashboard') );
-            });
+            // get dashboard layout from notebook-level dashboard metadata
+            var dashboardLayout = (notebook.metadata &&
+                                  notebook.metadata.urth &&
+                                  notebook.metadata.urth.dashboard &&
+                                  notebook.metadata.urth.dashboard.layout) ||
+                                  'no-layout';
```
Comment (parente) [inline routes/renderers.js]: We decided no layout metadata meant render as a report here. But in the extension we said if layout was missing, we'd look for grid metadata and assume grid is the default. Hmmm.

If I take a notebook that was produced by the 0.4.x extension and drop it in dashboard server now with this change, it's going to render in report layout and ignore all the grid metadata. That seems ... unfriendly.

Can we at least do a sniff test to see if any of the cells have the `urth.dashboard` path. If any of them do but layout is undefined in the notebook level metadata, default to grid. If there's no urth.dashboard anywhere then take the no-layout route.

Comment (dalogsdon) [inline routes/renderers.js]: I agree, and this is what @jameslmartin did in #145. I can reinstate that check.

Comment (parente) [inline routes/renderers.js]: OK. Cool. Just ping again when it's here and I'll finish testing.

**Comment by dalogsdon** [comment]
I updated the renderer to use grid layout if no layout was set but there is at least 1 cell that contains the grid metadata (row, col, width, height). I also noticed the Config script code in the dashboard template was always setting grid layout properties and not always putting in all default values if only some were set in the notebook. I've updated the template to handle these cases.

**Pull request merged**